docs/index.md

```diff
@@ -35,18 +35,73 @@ cd hello_world
 
 ## Run Your First Job
 
-Let's run our first job. It is pre-defined in the samples folder as part of the `init` command:
+Let's run our first job. It is pre-defined in the samples folder as part of the `init` command. Our job is going to load a CSV file from the samples folder and transform it into JSON:
 
 ```bash
 datayoga run sample.hello
 ```
 
-If all goes well, you should see some startup logs, and eventually:
+If all goes well, you should see a list of JSON values that correspond to each line in the input csv file:
 
 ```yaml
-{"greeting": "Hello Mr. John Doe"}
-{"greeting": "Hello Ms. Jane Doe"}
-{"greeting": "Hello Mr. Bill Adams"}
+{"id": "1", "fname": "john", "lname": "doe", "credit_card": "1234-1234-1234-1234", "country_code": "972", "country_name": "israel", "gender": "M", "full_name": "John Doe", "greeting": "HelMr. John Doe"}
+{"id": "2", "fname": "jane", "lname": "doe", "credit_card": "1000-2000-3000-4000", "country_code": "972", "country_name": "israel", "gender": "F", "full_name": "Jane Doe", "greeting": "HelMs. Jane Doe"}
+{"id": "3", "fname": "bill", "lname": "adams", "credit_card": "9999-8888-7777-666", "country_code": "1", "country_name": "usa", "gender": "M", "full_name": "Bill Adams", "greeting": "HelMr. Bill Adams"}
 ```
 
-That's it! You've created your first job that loads data from CSV, runs it through a series of transformation steps, and shows the data to the standard output. A good start. Read on for a more detailed tutorial or check out the [reference](reference/blocks.md) to see the different block types currently available.
+That's it! You've created your first job that loads data from CSV, runs it through a series of transformation steps, and shows the data to the standard output. A good start!
+
+## Enhance The Job Definition
+
+Let's customize the structure of the resulting JSON.
+The jobs are located under the `jobs` folder and can be arrange into modules. The `sample.hello` job you just ran is located in `jobs/sample/hello.yaml`.
+
+Add a new step of type `map` into the chain of steps. To do this, open the job definition in a text editor and add the section highlighted in bold to the job definition:
+
+<pre><code>
```

Comment:
this vanishes the YAML highlighting, what about keeping it (```yaml) and add a comment that stresses out the new section?

Reply:
will be hard to follow I think. Even this is not ideal. would be nicer if it was highlighted somehow...

Reply:
OK